resources/js/multisafepay.js

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+
+document.addEventListener('turbolinks:load', () => {
+    window.app.$on('checkout-payment-saved', (data) => {
+        if (!data.order.payment_method_code.includes('multisafepay_')) {
+            return;
+        }
+        window.app.checkout.doNotGoToTheNextStep = true
+
+        let headers = { Authorization: `Bearer ${localStorage.token}` }
+        if (window.config.store_code) {
+            headers['Store'] = window.config.store_code
+        }
+
+        axios.post(config.magento_url + '/graphql', {
+            query:
+            `query {
+                customer {
+                    orders ( filter: { number: { eq: "${ (data.order.id+'').padStart(9, '0') }" } } )
+                    {
+                        items {
+                            multisafepay_payment_url
+                        }
+                    }
+                }
+            }`,
```

Comment:
A query is going to be required to retrieve the `multisafepay_payment_url` from the order when you're checking out as a guest. The above works only when logged in.

I was not able to find a good workaround for this yet, as there doesn't seem to be a default query to retrieve an order like that, and the MSP plugins do not provide one.